Sources/HTTP/SimpleSocket/SimpleSocketConnectionListener.swift

```diff
@@ -137,17 +135,27 @@ public class BlueSocketConnectionListener: ParserConnecting {
             close()
         }
     }
+    
+    func cleanup() {
+        self.readerSource?.setEventHandler(handler: nil)
+        self.readerSource?.setCancelHandler(handler: nil)
+        
+        self.readerSource = nil
+        self.socket = nil
+        self.parser?.parserConnector = nil //allows for memory to be reclaimed
+        self.parser = nil
+    }
 
     /// Called by the parser to let us know that a response has started being created
     public func responseBeginning() {
-        self.socketWriterQueue?.async { [weak self] in
+        self.socketWriterQueue.async { [weak self] in
```

Comment:
I think all the calls to `async` here will add quite a bit of overhead. Can we not assume/require that all interaction done w/ the socket be done on the same serial queue?

Reply:
My concern here is to make sure that we can't get deadlocks. I've had a situation in testing where a read pushed a chunk of data into the pipeline, and it called something that wanted to write, but the write couldn't happen until the read completed, and so it all hung (this was on a future implementation in the context of SSL where OpenSSL's need to keep global state forces a lock around read/write). Maybe I'm just being paranoid. What do you think?